arkouda/pdarrayclass.py

```diff
@@ -606,149 +631,385 @@ def save(self, prefix_path, dataset='array', mode='truncate'):
             m = 0
         else:
             raise ValueError("Allowed modes are 'truncate' and 'append'")
-        rep_msg = generic_msg("tohdf {} {} {} {}".format(self.name, dataset, m, json.dumps([prefix_path])))
+
+        """
+        If offsets are provided, add to the json_array as the offsets will be used to 
+        retrieve the array elements from the hdf5 files.
+        """ 
+        try:
+            if offsets:
+                json_array = json.dumps([prefix_path, offsets])
```

Comment:
@reuster986 whoops, forgot to take this clause out. When a Strings object is created server-side, the offsets are returned back to the Python client. In the original design, the Strings save method was sending two arrays: (1) offsets AKA segments and (2) values. The user was not providing it (I agree, that would be confusing!), but rather the offsets are written as an instance attribute of the Strings class and were sent to server-side without the user knowing. And yep, to your second question, since pdarrayIO applies to pdarray and not Strings, I put a check in there for offsets.

Again, the current  design is just sending one array--the values array--to Chapel, so I'll remove this code as its no longer applicable.

Reply:
Fixed